app/post.go

```diff
@@ -28,14 +30,29 @@ func mdToHTML(md []byte) []byte {
 }
 
 func postHandler(c *gin.Context, app_settings common.AppSettings, database database.Database) ([]byte, error) {
+
+	// Get the post ID from the URL
+	postIDStr := c.Param("id")
```

Comment:
Use the `c.ShouldbindUri(...)` code instead of this, that does the integer conversion automatically.